### PR TITLE
test: convert a few more specs to async/await

### PR DESCRIPTION
**Pull request by miniak**
Notes: none

### CODE REVIEW

**Comment by jkleinsc** [comment]
@miniak try adding logging to see what rejection is happening, eg
```js
process.once('unhandledRejection', (reason, promise) => {
  console.log('Unhandled Rejection at:', promise, 'reason:', reason);
  done(new Error('catch block is delayed to next tick')));
}
```

File: spec/api-debugger-spec.ts [hunk 1]
```diff
@@ -29,13 +29,9 @@ describe('debugger module', () => {
...
     });
 
-    it('fails when protocol version is not supported', done => {
-      try {
-        w.webContents.debugger.attach('2.0');
-      } catch {
-        expect(w.webContents.debugger.isAttached()).to.be.false();
-        done();
-      }
+    it('fails when protocol version is not supported', async () => {
```
Comment (codebytere) [inline spec/api-debugger-spec.ts:32]: `w.webContents.debugger.attach('2.0')` throws, not rejects - I don't think there's any reason to have `async` here?

File: spec/spellchecker-spec.ts [hunk 2]
```diff
@@ -43,19 +43,18 @@ ifdescribe(features.isBuiltinSpellCheckerEnabled())('spellchecker', function ()
...
     // be found at src/third_party/hunspell_dictionaries/xx_XX.dic.
-    fs.readFile(path.join(__dirname, '/../../third_party/hunspell_dictionaries/xx-XX-3-0.bdic'), function (err, data) {
-      if (err) {
-        console.error('Failed to read dictionary file');
-        res.writeHead(404);
-        res.end(JSON.stringify(err));
-        return;
-      }
+    try {
+      const data = fs.readFile(path.join(__dirname, '/../../third_party/hunspell_dictionaries/xx-XX-3-0.bdic'));
```
Comment (codebytere) [inline spec/spellchecker-spec.ts:50]: If we're using `fs.promises` here we need to `await` the call to `fs.readFile`

Comment (miniak) [reply spec/spellchecker-spec.ts]: missed that, thanks!

File: spec/api-debugger-spec.ts (hunk 1 above)
Comment (miniak) [reply spec/api-debugger-spec.ts:32]: I have mirrored the existing
https://github.com/electron/electron/blob/7981d955b8f84e2129f5a907445fbc0aea6b9fe8/spec/api-debugger-spec.ts#L41-L44

Comment (miniak) [reply spec/api-debugger-spec.ts]: fixing both

**Review by codebytere — APPROVED** [review]

**Pull request merged**